pkg/odo/cli/preference/registry/util/util.go

```diff
@@ -28,10 +32,10 @@ func IsSecure(prefClient preference.Client, registryName string) bool {
 	return isSecure
 }
 
-func IsGitBasedRegistry(url string) bool {
-	return strings.Contains(url, "github.com") || strings.Contains(url, "raw.githubusercontent.com")
-}
-
-func PrintGitRegistryDeprecationWarning() {
-	log.Deprecate("Git based registries", "Please see https://github.com/redhat-developer/odo/tree/main/docs/public/git-registry-deprecation.adoc")
+func IsGithubBasedRegistry(url string) (bool, error) {
+	pu, err := url2.Parse(url)
+	if err != nil {
+		return false, fmt.Errorf("unable to parse registry url %w", err)
+	}
+	return strings.Contains(pu.Host, "github.com") || strings.Contains(url, "raw.githubusercontent.com"), nil
```

Comment:
This would still return `true` for non-GitHub URLs that are in the `.com` TLD and end with `github` or `raw.githubusercontent`, like `https://my.devfile.registry-from-github.com`.
I think we should test that the host part is not part of the real GitHub domains.
Something like this maybe:
```go
	return pu.Host == "github.com" || strings.HasSuffix(pu.Host, ".github.com") ||
		pu.Host == "raw.githubusercontent.com" || strings.HasSuffix(pu.Host, ".raw.githubusercontent.com"), nil
```